python/ray/ml/examples/horovod/horovod_pytorch_example.py

```diff
@@ -0,0 +1,266 @@
+import argparse
+from filelock import FileLock
+import horovod.torch as hvd
+import os
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+import torch.utils.data.distributed
+from torchvision import datasets, transforms
+
+import ray
+from ray import train
+from ray.ml.train.integrations.horovod import HorovodTrainer
+
+
+def metric_average(val, name):
+    tensor = torch.tensor(val)
+    avg_tensor = hvd.allreduce(tensor, name=name)
+    return avg_tensor.item()
+
+
+class Net(nn.Module):
+    def __init__(self):
+        super(Net, self).__init__()
+        self.conv1 = nn.Conv2d(1, 10, kernel_size=5)
+        self.conv2 = nn.Conv2d(10, 20, kernel_size=5)
+        self.conv2_drop = nn.Dropout2d()
+        self.fc1 = nn.Linear(320, 50)
+        self.fc2 = nn.Linear(50, 10)
+
+    def forward(self, x):
+        x = F.relu(F.max_pool2d(self.conv1(x), 2))
+        x = F.relu(F.max_pool2d(self.conv2_drop(self.conv2(x)), 2))
+        x = x.view(-1, 320)
+        x = F.relu(self.fc1(x))
+        x = F.dropout(x, training=self.training)
+        x = self.fc2(x)
+        return F.log_softmax(x)
+
+
+def setup(config):
+    data_dir = config.get("data_dir", None)
+    seed = config.get("seed", 42)
+    batch_size = config.get("batch_size", 64)
+    use_adasum = config.get("use_adasum", False)
+    lr = config.get("lr", 0.01)
+    momentum = config.get("momentum", 0.5)
+    use_cuda = config.get("use_cuda", False)
```

Comment:
@amogkam @matthewdeng 
Seems like config should really not contain `use_cuda` here. It causes confusion. `use_gpu` should be only contained in ScalingConfig.
Should we consider removing `use_cuda` here and solely rely on:

```
device = get_device()
if device.type.startswith("cuda"):
  # xxx.to(device) should be called here.
```

Reply:
This is just the user config- they can pass in whatever information they want here to use inside the `train_loop_per_worker`.

You could even specify a `use_gpu=True` at the start of the file and then directly access the same variable for both the scaling config and inside the training function.

Reply:
ok, I see. 
Let's keep it then.